docs/rules/prefer-one-generic-in-create-for-feature-selector.md

```diff
@@ -0,0 +1,24 @@
+# Prefer One Generic In `createForFeatureSelector`
+
+> Prefer using a single generic to define the feature state.
+
+## Rule Details
+
+`createFeatureSelector` is typically used with `forFeature`, which should not be aware about the shape of the Global Store. Most of the time, feature states are lazy-loaded. As such, they only need to know (and care) about their own shape.
```

Comment:
is it too much to touch also the selector topic? regarding composing?

Reply:
Did you had something in mind to add here?

Reply:
side note: coming from the origin discussion with david, for me it's still not clear (had no time to check it by myself) what happens if i compose a selector from lazy state A (not yet loaded) with a selector from lazy state B (already loaded).

"createFeatureSelector from a not loaded lazy state composed with another createFeatureSelector from a loaded lazy state throws an error"
or
"createFeatureSelector from a not loaded lazy state composed with another createFeatureSelector from a loaded lazy state emits data only when all state is loaded"

very bad examples but i hope you understand what im trying to say.

Reply:
I see, the returned state is `undefined` from `createFeatureSelector`. 
In most cases that I've experienced, this results in errors because the selectors that use this state are accessing properties on an unexciting state.

Reply:
I've added a note about this (https://github.com/timdeschryver/eslint-plugin-ngrx/pull/157/commits/c82b1a3ebb8daaa6c67ed54f088724629de5a438), do you think this is clear enough, or does it need something more?

Reply:
it's great! thank you!💪🏻